core/execution/special_orders.go

```diff
@@ -268,57 +248,183 @@ func (m *Market) updateLPOrders(
-				distressedParties[order.Party] = struct{}{}
-			} else if len(conf.Trades) > 0 {
-				m.log.Panic("submitting liquidity orders after a reprice should never trade",
-					logging.Order(*order))
+			order.Status = types.OrderStatusCancelled
+			orderEvts = append(orderEvts, events.NewOrderEvent(ctx, order))
+			continue
+		}
+
+		subFn(order)
+	}
+
+	for _, order := range submits {
+		order.UpdatedAt = now
+		subFn(order)
+	}
+
+	// send cancel events
+	m.broker.SendBatch(orderEvts)
+
+	// an ordered list of positions
+	var (
+		positions     = make([]events.MarketPosition, 0, len(partiesPos))
+		marginsBefore = map[string]*num.Uint{}
+		id            = m.GetID()
+		assetID, _    = m.mkt.GetAsset()
+	)
+	// now we can check parties positions
+	for party, pos := range partiesPos {
+		positions = append(positions, pos)
+		mar, _ := m.collateral.GetPartyMarginAccount(id, party, assetID)
+		marginsBefore[party] = mar.Balance
+	}
+
+	sort.Slice(positions, func(i, j int) bool {
+		return positions[i].Party() < positions[j].Party()
+	})
+
+	// now we calculate all the new margins
+	risks := m.updateMargin(ctx, positions)
+	if len(risks) > 0 {
+		transfers, closed, bondPenalties, err := m.collateral.MarginUpdate(
+			ctx, m.GetID(), risks)
+		if err == nil && len(transfers) > 0 {
+			evt := events.NewLedgerMovements(ctx, transfers)
+			m.broker.Send(evt)
+		}
+
+		cancelled := m.applyBondPenaltiesAndLiquidationExcludingPending(
+			ctx, bondPenalties, closed, marginsBefore,
+		)
+
+		// now ensure we have all parties pending status updated
+		for _, v := range positions {
+			if m.liquidity.IsLiquidityProvider(v.Party()) {
+				if _, ok := cancelled[v.Party()]; !ok {
+					// this party LP wasn't cancelled, so it should be now
+					// not pending anymore,
+					m.liquidity.RemovePending(v.Party())
+				}
 			}
 		}
 
-		// if an auction has been started, we just break now
-		if m.as.InAuction() {
-			// enteredAuction = true
-			// auctionFrom = i
-			// break
-			return nil
+		_ = m.equityShares.SharesExcept(m.liquidity.GetInactiveParties())
+
+		m.updateLiquidityFee(ctx)
+	}
+
+	return []*types.Order{}
+}
+
+func (m *Market) applyBondPenaltiesAndLiquidationExcludingPending(
+	ctx context.Context,
+	bondPenalties []events.Margin,
+	closed []events.Margin,
+	initialMargins map[string]*num.Uint,
+) map[string]struct{} {
+	var (
+		cancelled    = map[string]struct{}{}
+		reallyClosed = []events.Margin{}
+	)
+
+	// alright, here we need to go weird over things because we want to find what
+	// parties have been considered distressed by the risk / collateral engine BUT
+	// for which the LP submission where still pending a first deployment.
+	// In which case no bond slashing is being taken, and they shall not be
+	// closed as well but the lp submission should only be cancelled.
+
+	// so first we will find all pending which would be closed
+	for _, v := range closed {
+		if m.liquidity.IsPending(v.Party()) {
+			_ = m.cancelPendingLiquidityProvision(
```

Comment:
err here? does it matter?

Reply:
nope doesnt' mattert here, it cannot fail, API needs to be fixed later to reflect this.